extensions/cxf/deployment/src/main/java/io/quarkus/cxf/deployment/CxfProcessor.java

```diff
@@ -0,0 +1,136 @@
+package io.quarkus.cxf.deployment;
+
+import java.util.Map.Entry;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.Type;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanGizmoAdaptor;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.cxf.runtime.AbstractCxfWebServiceProducer;
+import io.quarkus.cxf.runtime.CXFQuarkusServlet;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.util.HashUtil;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.undertow.deployment.ServletBuildItem;
+
+/**
+ * Processor that finds CXF web service classes in the deployment
+ */
+public class CxfProcessor {
+
+    private static final String JAX_WS_CXF_SERVLET = "org.apache.cxf.transport.servlet.CXFNonSpringServlet;";
+
+    private static final DotName WEBSERVICE_ANNOTATION = DotName.createSimple("javax.jws.WebService");
+    private static final DotName WEBSERVICE_PRODUCER = DotName.createSimple(AbstractCxfWebServiceProducer.class.getName());
+
+    @BuildStep
+    UnremovableBeanBuildItem markBeansAsUnremovable() {
```

Comment:
@Dufgui maybe we can remove this because we have ever a UnremovableBeanBuildItem  produce lower

Reply:
Yes